canister/src/validation.rs

```diff
@@ -1,11 +1,11 @@
-use crate::{blocktree::BlockDoesNotExtendTree, state::State, unstable_blocks, Block};
+use crate::{blocktree::BlockDoesNotExtendTree, state::State, unstable_blocks};
 use bitcoin::BlockHeader;
 use ic_btc_validation::HeaderStore;
 
 /// A structure passed to the validation crate to validate a specific block header.
 pub struct ValidationContext<'a> {
     state: &'a State,
-    chain: Vec<&'a Block>,
+    chain: Vec<(&'a BlockHeader, crate::types::BlockHash)>,
```

Comment:
nit: Please add a comment as to why we're storing both the has and header, and maybe add a jira ticket as well for us to clean this up later.